### PR TITLE
Keep ProvisionDockServer service consistent with ProvisionControllerServer service

### PR DESCRIPTION
**Pull request by leonwanghui**
Signed-off-by: leonwanghui <wanghui71leon@gmail.com>

**What this PR does / why we need it**:
This PR is proposed to add `GetMetrics` and `GetUrls` method in ProvisionDockServer service so as to keep consistent with ProvisionControllerServer service. Besides, some field validation code is also added in apiserver module.

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #

**Special notes for your reviewer**:

**Release note**:

```release-note
```

### CODE REVIEW

**Comment by leonwanghui** [comment]
@xing-yang @wisererik PTAL, thanks!

File: pkg/api/controllers/profile.go [hunk 1]
```diff
@@ -64,7 +64,7 @@ func (p *ProfilePortal) CreateProfile() {
 				profile.ProvisioningProperties.DataStorage.StorageAccessCapability = []string{"Read", "Write", "Execute"}
 			}
 		}
-		if io := pp.IOConnectivity; io.IsEmpty(){
+		if io := pp.IOConnectivity; io.IsEmpty() {
```
Comment (skdwriting) [inline pkg/api/controllers/profile.go]: just a line position change?! no code change..

File: pkg/api/util/db.go [hunk 2]
```diff
@@ -120,6 +125,12 @@ func CreateFileShareDBEntry(ctx *c.Context, in *model.FileShareSpec) (*model.Fil
 	if in.Id == "" {
 		in.Id = uuid.NewV4().String()
 	}
+	// validate profileId
+	if in.ProfileId == "" {
```
Comment (skdwriting) [inline pkg/api/util/db.go]: really need this duplicate check code? cannot be avoided?

File: pkg/dock/dock.go [hunk 3]
```diff
@@ -585,6 +584,18 @@ func (ds *dockServer) DeleteFileShareSnapshot(ctx context.Context, opt *pb.Delet
...
 		return pb.GenericResponseError(err), err
 	}
-	// TODO: maybe need to update status in DB.
+
 	return pb.GenericResponseResult(nil), nil
 }
+
+// GetMetrics method is only defined to make ProvisioinDock service consistent with
+// ProvisionController service, so this method is not allowed to be called.
+func (ds *dockServer) GetMetrics(context.Context, *pb.GetMetricsOpts) (*pb.GenericResponse, error) {
```
Comment (skdwriting) [inline pkg/dock/dock.go]: Why do we need these two functions just to send the message that it is not implemented?

File: pkg/model/proto/model.proto [hunk 4]
```diff
@@ -147,6 +147,12 @@ service ProvisionDock {
 
     // Collect metrics from metrics driver
     rpc CollectMetrics (CollectMetricsOpts) returns (GenericResponse){}
+
+    // Get metrics from Prometheus
+    rpc GetMetrics (GetMetricsOpts) returns (GenericResponse){}
```
Comment (skdwriting) [inline pkg/model/proto/model.proto]: purpose?! same for GetUrls

File: pkg/api/controllers/profile.go (hunk 1 above)
Comment (leonwanghui) [reply pkg/api/controllers/profile.go]: That's the impact after running goimports command.

File: pkg/dock/dock.go (hunk 3 above)
Comment (leonwanghui) [reply pkg/dock/dock.go]: If you look at [comments](https://github.com/opensds/opensds/pull/888#discussion_r294332438) from xing, we came to conclusion that we should keep thin-opensds as consistent as possible with master branch.

File: pkg/model/proto/model.proto (hunk 4 above)
Comment (leonwanghui) [reply pkg/model/proto/model.proto]: Please see the comment above.

**Comment by xing-yang** [comment]
Please rebase.

**Comment by leonwanghui** [comment]
@xing-yang Please take a review, thanks!

**Comment by xing-yang** [comment]
@leonwanghui The changes look good to me.  Can you please squash your commits into one?  Thanks.

**Pull request closed without merging**